### PR TITLE
add DL test

### PR DESCRIPTION
**Pull request by yito88**
https://scalar-labs.atlassian.net/browse/DLT-2312

- [x] Setup DB and DL server
- [x] Add the runner
- [x] Add a simple test
- [x] Add nemeses
- [x] Switch a server
- [x] Handle an unknown failure

I've added a `simple` test. I'll add `transfer` test at another PR.

### CODE REVIEW

File: scalardl/src/scalardl/simple.clj [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                (do
+                  (reset! client-service
+                          (dl/try-switch-server! @client-service test))
+                  (assoc op :type :fail :error (.getMessage resp)))))
+
+      :write (let [v (:value op)
+                   argument (create-argument 1 v)
+                   resp (.executeContract @client-service "write" argument)]
+               (if (or (util/success? resp)
+                       (and (util/unknown? resp)
```
Comment (feeblefakie) [inline scalardl/src/scalardl/simple.clj]: As we discussed in person, maybe we can throw `info` after some retries failed ?

File: scalardl/src/scalardl/simple.clj [hunk 2]
```diff
@@ -0,0 +1,124 @@
+(ns scalardl.simple
```
[the remark this answers is not in the document]
Comment (yito88) [reply scalardl/src/scalardl/simple.clj]: Thanks! Renamed to `cas`.

File: scalardl/src/scalardl/simple.clj (hunk 1 above)
Comment (yito88) [reply scalardl/src/scalardl/simple.clj]: Thank you for your proposal!
Changed the handling the unknown status.

File: scalardl/contract/Cas.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    int value = argument.getInt("value");
+    if (current != value) {
+      throw new ContractContextException(
+          "The current value isn't the same as the value in the argument");
+    }
+
+    int next = argument.getInt("new_value");
+    ledger.put(key, Json.createObjectBuilder().add("value", next).build());
+
+    return null;
```
[the remark this answers is not in the document]
Comment (feeblefakie) [reply scalardl/contract/Cas.java]: Sorry it's a lie ! I was confused...

**Pull request merged**